Add support for CakePHP 5

ishanvyas22 opened the pull request:
ref. https://github.com/bcrowe/cakephp-api-pagination/issues/14

@bcrowe The PR can be merged after you move current code into 3.x branch for CakePHP 4 support.

For the support of PHP 8.4, it needs to have CakePHP 5.1 which blocks the users who wants to use cake 5.0. So it's better to include it into a separate minor release.

ishanvyas22 committed:
Add CakePHP 5 support

ishanvyas22 committed:
Drop support PHP < 8.1

ishanvyas22 committed:
Fix CI

ishanvyas22 committed:
Fix all failing tests

ishanvyas22 committed:
Remove PHP 8.4

PHP 8.4 support isn't possible with cakephp 5.0

bcrowe commented:
🙌  Thanks for this @ishanvyas22 ! I'll take a look over this later tonight and get things merged and make a release.

ishanvyas22 commented:
Hey @bcrowe, did you had a chance to look at it? No pressure just following up.

ishanvyas22 committed:
Keep paging attribute check

If paging attribute is present in the request object it is given precedence over PaginatedInterface object.

PaulHendriks commented:
What's the status for this PR?
I just did a fresh CakePHP 5.2.7 install and wanted to implement this plugin.
Composer rejected it because of the CakePHP4 dependencies.
